### PR TITLE
feat(password): remove reset password tokens on change password

### PR DESCRIPTION
**Pull request by pfabrikant**
Adding the method `removeAllPasswordResetTokens` to the `password`  package. This method is called within the `changePassword` method. It removes all password reset tokens from the user storage.

### CODE REVIEW

File: packages/password/src/accounts-password.ts [hunk 1]
```diff
@@ -88,6 +88,12 @@ export interface AccountsPasswordOptions {
    * Will automatically send a verification email after signup.
    * Default to false.
    */
+
+  removeAllPasswordResetTokensAfterPasswordChanged?: boolean;
+  /**
```
Comment (pradel) [inline packages/password/src/accounts-password.ts:93]: looks like this comment is not for the good option ;)

File: packages/password/__tests__/accounts-password.ts [hunk 2]
```diff
@@ -397,10 +431,12 @@ describe('AccountsPassword', () => {
       const userId = 'id';
       const setPassword = jest.fn(() => Promise.resolve('user'));
       const findUserById = jest.fn(() => Promise.resolve(validUser));
-      password.setStore({ setPassword, findUserById } as any);
+      const removeAllPasswordResetTokens = jest.fn(() => Promise.resolve());
```
Comment (pradel) [inline packages/password/__tests__/accounts-password.ts:434]: in this test, where do you check that it was actually called?

File: packages/database-mongo/src/mongo.ts [hunk 3]
```diff
@@ -364,6 +364,18 @@ export class Mongo implements DatabaseInterface {
     );
   }
 
+  public async removeAllPasswordResetTokens(userId: string): Promise<void> {
+    const id = this.options.convertUserIdToMongoObjectId ? toMongoID(userId) : userId;
+    await this.collection.update(
+      { _id: id },
+      {
+        $set: {
```
Comment (pradel) [inline packages/database-mongo/src/mongo.ts:372]: I am not sure it really makes sense to set an empty array here, we could just unset the value to leave a clean object in the db, what do you think?

File: packages/database-mongo/__tests__/index.ts [hunk 4]
```diff
@@ -725,6 +725,30 @@ describe('Mongo', () => {
     });
   });
 
+  describe('removeAllPasswordResetTokens', () => {
+    // eslint-disable-next-line jest/expect-expect
+    it('should not convert id', async () => {
+      const mongoOptions = new Mongo(databaseTests.db, {
+        convertUserIdToMongoObjectId: false,
+      });
+      await mongoOptions.setService('toto', 'twitter', { id: '1' });
```
Comment (pradel) [inline packages/database-mongo/__tests__/index.ts:734]: you need to call `removeAllPasswordResetTokens` here 😁

File: packages/database-mongo/__tests__/index.ts [hunk 5]
```diff
@@ -725,6 +725,30 @@ describe('Mongo', () => {
...
+    });
+
+    it('should remove the password reset tokens', async () => {
```
Comment (pradel) [inline packages/database-mongo/__tests__/index.ts:737]: perfect test 👌

File: packages/password/src/accounts-password.ts (hunk 1 above)
Comment (pfabrikant) [reply packages/password/src/accounts-password.ts:93]: oops :)

File: packages/password/__tests__/accounts-password.ts (hunk 2 above)
Comment (pfabrikant) [reply packages/password/__tests__/accounts-password.ts:434]: My mistake. I checked the wrong method in the assertion.

File: packages/database-mongo/__tests__/index.ts (hunk 4 above)
Comment (pfabrikant) [reply packages/database-mongo/__tests__/index.ts:734]: 🤦‍♂

File: packages/password/src/accounts-password.ts [hunk 6]
```diff
@@ -84,6 +84,11 @@ export interface AccountsPasswordOptions {
    * Default to false.
    */
   invalidateAllSessionsAfterPasswordChanged?: boolean;
+  /**
+   * Will remove all password reset tokens from the db after a password has been changed.
+   * Default to false.
```
Comment (pradel) [inline packages/password/src/accounts-password.ts:89]: default to true actually :)

Comment (pfabrikant) [reply packages/password/src/accounts-password.ts:89]: 🤦‍♂

**Comment by pradel** [comment]
Thanks @pfabrikant! Will create now

**Pull request merged**